ons-spark/raw-notebooks/checkpoint-staging/r_input.R

```diff
@@ -0,0 +1,155 @@
+options(warn = -1) 
+library(sparklyr)
+library(dplyr)
+library(DBI)
+
+sc <- sparklyr::spark_connect(
+    master = "local[2]",
+    app_name = "checkpoint",
+    config = sparklyr::spark_config())
+
+
+set.seed(42)
+new_cols <- 12
+num_rows <- 10^3
+
+start_time <- Sys.time()
+
+df = sparklyr::sdf_seq(sc, 1, num_rows) %>%
+    sparklyr::mutate(col_0 = ceiling(rand()*new_cols))
+
+for (i in 1: new_cols)
+{
+  column_name = paste0('col_', i)
+  prev_column = paste0('col_', i-1)
+  df <- df %>%
+    sparklyr::mutate(
+      !!column_name := case_when(
+        !!as.symbol(prev_column) > i ~ !!as.symbol(prev_column)))
```

Comment:
I see what you mean about getting the nulls and not the zeroes in the tables – I was also getting errors when trying to change it to a zero. I’m not sure why exactly this worked (I’m sure I tried the same thing before and got an error) but in the end changing the for loop to this worked: 

```
for (i in 1: new_cols)
{
  column_name <- paste0('col_', i)
  prev_column <- paste0('col_', i-1)
  df <- df %>%
    mutate(
      !!column_name := case_when(
        !!as.symbol(prev_column) > i ~ !!as.symbol(prev_column),
        TRUE ~ 0)) 
}
```

(I added a “, TRUE ~0” as another argument to case_when). I'm sure you probably tried this so not sure why it's working for me now - but it seems to be running absolutely fine now. Can share more of my code if it's still erroring for you.

Reply:
Oh - obviously make sure it's changed in the checkpointing for loop too (though I'm sure you would've done!)